timelock-agent/src/main/java/com/palantir/timelock/TimeLockAgent.java

```diff
@@ -64,39 +62,22 @@ public TimeLockAgent(
      * @param client Client namespace to create the services for
      * @return Invalidating timestamp and lock services
      */
-    abstract TimeLockServices createInvalidatingTimeLockServices(String client);
-
-    public void createAndRegisterResources() {
-        // TODO: actually implement this correctly, and not just to make it compile
-        runtime.map(config -> config.clients());
-        Map<String, TimeLockServices> clientToServices = ImmutableMap.copyOf(Maps.asMap(
-                blockingMostRecent(runtime.map(TimeLockRuntimeConfiguration::clients)).get(),
-                client -> createInvalidatingTimeLockServices(client)));
-
-        registrar.accept(new TimeLockResource(clientToServices));
-    }
-
-    // copied the below from WC internally
-    // will need this functionality in a utility class somewhere
+    public abstract TimeLockServices createInvalidatingTimeLockServices(String client);
```

Comment:
Yep, good spot. The public API here should really only be `createAndRegisterResources()`.